baybe/simulation/lookup.py

```diff
@@ -70,41 +71,20 @@ def look_up_targets(
     if lookup is None:
         add_fake_measurements(queries, targets)
     elif isinstance(lookup, Callable):
-        _look_up_targets_from_callable(queries, targets, lookup)
+        _look_up_targets_from_callable(queries, lookup)
     elif isinstance(lookup, pd.DataFrame):
         _look_up_targets_from_dataframe(queries, targets, lookup, impute_mode)
     else:
         raise ValueError("Unsupported lookup mechanism.")
 
 
 def _look_up_targets_from_callable(
-    queries: pd.DataFrame,
-    targets: Collection[Target],
-    lookup: Callable,
+    queries: pd.DataFrame, lookup: Callable[[pd.DataFrame], pd.DataFrame]
 ) -> None:
     """Look up target values by querying a callable."""
```

Comment:
Although clear from the type hints, a small "Modifies `queries` in-place" would be nice.

Reply:
True, but this needs to be actually changed in many places in the simulation module (e.g. also the other lookup mechanism). Let me postpone these cosmetics to the point when the actual remaining refactoring changes are done for the simulation module, otherwise we'll probably do unnecessary work here and there 🙃 Once the logic is bullet-proof, I'll revise the docstrings anyway 👍🏼